app/src/main/java/com/bnyro/recorder/util/MediaTrimmer.kt

```diff
@@ -46,14 +49,25 @@ class MediaTrimmer {
         inputPfd: ParcelFileDescriptor,
         outputPfd: ParcelFileDescriptor,
         startMs: Long,
-        endMs: Long
+        endMs: Long,
+        extension: String?
     ): Boolean {
         return withContext(Dispatchers.IO) {
             val extractor = MediaExtractor()
             extractor.setDataSource(inputPfd.fileDescriptor)
             val trackCount = extractor.trackCount
+            var outputFormat = MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4
+            if (extension == "3gp") {
+                outputFormat = MediaMuxer.OutputFormat.MUXER_OUTPUT_3GPP
+            }
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q && extension == "ogg") {
+                outputFormat = MediaMuxer.OutputFormat.MUXER_OUTPUT_OGG
+            }
+            if (extension == "webm") { // No need to add API level 21 here as the whole app is minSdk 21
+                outputFormat = MediaMuxer.OutputFormat.MUXER_OUTPUT_WEBM
+            }
```

Comment:
Please simplify this by using a `when` statement here.

Reply:
I don't know how to also test for the API version for ogg like this...
```
            val extractor = MediaExtractor()
            extractor.setDataSource(inputPfd.fileDescriptor)
            val trackCount = extractor.trackCount
            val muxer = MediaMuxer(
                outputPfd.fileDescriptor,
                when (extension) {
                    "3gp" -> MediaMuxer.OutputFormat.MUXER_OUTPUT_3GPP
                    "ogg" -> MediaMuxer.OutputFormat.MUXER_OUTPUT_OGG // How do I also handle (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q)
                    "webm" -> MediaMuxer.OutputFormat.MUXER_OUTPUT_WEBM
                    else -> MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4
                }
            )

```

Reply:
`"ogg" -> if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) MediaMuxer.OutputFormat.MUXER_OUTPUT_OGG else MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4`

Reply:
I've end up tweaking the extension as well otherwise we would get MP4 container format saved into a file with .ogg extension. Can you please accept it like this?

Reply:
Fine with me :+1:

Please remove the comment `// How do I handle ...` though